### PR TITLE
no_log: more false-positives (not flagged by sanity tests yet)

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Flagged by ansible/ansible#73882.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
keycloak
linode_v4
proxmox_kvm
oneandone_server
profitbricks
profitbricks_volume
sl_vm
aerospike_migrations
datadog_event
pagerduty_alert
manageiq_provider
gitlab_user

### CODE REVIEW

**Review by mcltn — APPROVED** [review]

**Review by briantist — APPROVED** [review]

File: plugins/modules/cloud/misc/proxmox_kvm.py [hunk 1]
```diff
@@ -1057,7 +1057,7 @@ def main():
             smbios=dict(type='str'),
             snapname=dict(type='str'),
             sockets=dict(type='int'),
-            sshkeys=dict(type='str'),
+            sshkeys=dict(type='str', no_log=False),
```
Comment (Aversiste) [inline plugins/modules/cloud/misc/proxmox_kvm.py:1060]: This is not very useful for proxmox_kvm as this parameter is actually a public key, and therefore not a secret.

File: plugins/modules/cloud/profitbricks/profitbricks.py [hunk 2]
```diff
@@ -583,7 +583,7 @@ def main():
             volume_size=dict(type='int', default=10),
             disk_type=dict(choices=['HDD', 'SSD'], default='HDD'),
             image_password=dict(default=None, no_log=True),
-            ssh_keys=dict(type='list', elements='str', default=[]),
+            ssh_keys=dict(type='list', elements='str', default=[], no_log=False),
```
Comment (rndmh3ro) [inline plugins/modules/cloud/profitbricks/profitbricks.py:586]: Is this correct? The dict has the private key as a key:https://github.com/ansible-collections/community.general/blob/3053a85e567fd5a70e6801b464f04f9b5b266672/plugins/modules/remote_management/manageiq/manageiq_provider.py#L174
ir does this not get logged?

File: plugins/modules/source_control/gitlab/gitlab_user.py [hunk 3]
```diff
@@ -470,7 +470,7 @@ def main():
         password=dict(type='str', no_log=True),
         email=dict(type='str'),
         sshkey_name=dict(type='str'),
-        sshkey_file=dict(type='str'),
+        sshkey_file=dict(type='str', no_log=False),
```
Comment (rndmh3ro) [inline plugins/modules/source_control/gitlab/gitlab_user.py:473]: This seems to be a private key: https://github.com/ansible-collections/community.general/blob/3053a85e567fd5a70e6801b464f04f9b5b266672/plugins/modules/source_control/gitlab/gitlab_user.py#L62

File: plugins/modules/cloud/profitbricks/profitbricks.py (hunk 2 above)
Comment (rndmh3ro) [inline plugins/modules/cloud/profitbricks/profitbricks.py:586]: Is this correct? The dict contains the private key: https://github.com/ansible-collections/community.general/blob/3053a85e567fd5a70e6801b464f04f9b5b266672/plugins/modules/remote_management/manageiq/manageiq_provider.py#L174
Or does the key not get logged?

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 1 above)
Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox_kvm.py:1060]: Yes, that's why `no_log` is set to `False`, to indicate that it is not secret.

File: plugins/modules/cloud/profitbricks/profitbricks.py (hunk 2 above)
Comment (felixfontein) [reply plugins/modules/cloud/profitbricks/profitbricks.py:586]: `auth_key` itself is marked as `no_log=True`: https://github.com/ansible-collections/community.general/blob/3053a85e567fd5a70e6801b464f04f9b5b266672/plugins/modules/remote_management/manageiq/manageiq_provider.py#L593

File: plugins/modules/source_control/gitlab/gitlab_user.py (hunk 3 above)
Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_user.py:473]: From the example (and from how GitLab/GitHub work), it is a public key: https://github.com/ansible-collections/community.general/blob/3053a85e567fd5a70e6801b464f04f9b5b266672/plugins/modules/source_control/gitlab/gitlab_user.py#L126

File: plugins/modules/cloud/profitbricks/profitbricks.py (hunk 2 above)
Comment (felixfontein) [reply plugins/modules/cloud/profitbricks/profitbricks.py:586]: (Same as above, I'll mark this as resolved.)

File: plugins/modules/source_control/gitlab/gitlab_user.py (hunk 3 above)
Comment (rndmh3ro) [reply plugins/modules/source_control/gitlab/gitlab_user.py:473]: Sorry, you're right!

File: plugins/modules/cloud/misc/proxmox_kvm.py (hunk 1 above)
Comment (Aversiste) [reply plugins/modules/cloud/misc/proxmox_kvm.py:1060]: Oops, I apparently stopped reading after seeing the parameter `no_log`.

Comment (felixfontein) [reply plugins/modules/cloud/misc/proxmox_kvm.py:1060]: :D That does happen ;-) Thanks for checking!

File: plugins/modules/source_control/gitlab/gitlab_user.py (hunk 3 above)
Comment (felixfontein) [reply plugins/modules/source_control/gitlab/gitlab_user.py:473]: No problem, thanks a lot for checking :) I tried to check all of them, but I don't know most of the modules and am thus very happy if someone else takes a look as well (and in particular when it turns out I didn't screw it up ;) )!

**Review by aminvakil — APPROVED** [review]

**Comment by felixfontein** [comment]
@mcltn @briantist @Aversiste @rndmh3ro @aminvakil thanks all for reviewing this! :)

**Comment by felixfontein** [comment]
I created a backport to stable-1 manually in #2017. I added fixes for four other modules (one is a 'backport' from community.docker, one is for a module moved to community.google, and for two modules the type of the option was changed to `path` in 2.x.0, and is only left to `str` since we don't backport features).

**Pull request merged**